HARK/core.py

```diff
@@ -513,6 +514,17 @@ def initialize_sim(self):
             self.AgentCount, dtype=int
         )  # Which cycle period each agent is on
         self.sim_birth(all_agents)
+
+        # If we are asked to use existing shocks and a set of initial conditions
+        # exist, use them
+        if self.read_shocks and bool(self.newborn_init_history):
+
+            for var_name in self.state_now:
+                # Copy only array-like idiosyncratic states. Aggregates should
+                # not be set by newborns
+                if hasattr(self.state_now[var_name], "shape"):
```

Comment:
I am a little skeptical about using `hasattr(*,"shape")`, which you use a few times in this PR, as the test for being array-like.

Are we being so inconsistent in the code that a proper type check won't work here?

e.g `isinstance(self.state_now[var_name], np.ndarray)`

Reply:
Fair, and better. Thank you!

An issue that I have yet to resolve is that in some models (`ConsIndShock`) an aggregate state like `PLvlAgg` will be a float, but in others (like `PortfolioFrame`) it will be an `np.array` of length 1. So yes we are being inconsistent!

Reply:
@sbenthall, I wonder if it is your view that we should never have used floats at all but instead `np.array` of length 1 pervasively throughout, and you are trying to move us in that direction with new code you write?  I might buy this as a reasonable proposition if it were articulated ...

Reply:
I see. I hadn't realized I had introduced that inconsistency.

@llorracc in terms of my preferred direction...

...in the Frame model code, it's the frame that carries the information that its variables are aggregate. This gets used directly by the simulation code instead of doing a type check on the state value:

https://github.com/econ-ark/HARK/blob/master/HARK/ConsumptionSaving/ConsPortfolioFrameModel.py#L189-L192

https://github.com/econ-ark/HARK/blob/master/HARK/frame.py#L112-L117